ipn/ipnlocal/local.go

```diff
@@ -4093,3 +4006,34 @@ func (b *LocalBackend) SetDevStateStore(key, value string) error {
 func (b *LocalBackend) ShouldInterceptTCPPort(port uint16) bool {
 	return b.shouldInterceptTCPPortAtomic.Load()(port)
 }
+
+func (b *LocalBackend) SwitchProfile(profile ipn.ProfileID) error {
+	if b.CurrentProfile().ID == profile {
+		return nil
+	}
+	b.enterState(ipn.NoState) // Reset state.
```

Comment:
May want to do an early check that the profile is known, otherwise you'll mutate the state but return an error.

Reply:
The lock ordering makes it slightly tricky. Done.